File: 20210719/괄호/괄호.js
function solution(expression) {
  let expressionList = expression.split("-");
  let total = 0;

  for (let i = 0; i < expressionList.length; i++) {
    if (i == 0) {
      total += expressionList[i]
        .split("+")
        .reduce((acc, cur) => parseInt(acc) + parseInt(cur));
    } else {
      total -= expressionList[i]
        .split("+")
        .reduce((acc, cur) => parseInt(acc) + parseInt(cur));
    }
  }

  console.log(total);
}

solution("55-50+40");

File: 20210719/괄호/괄호.py
expression = input().split(sep='-')
total = 0
for idx, value in enumerate(expression):
    if idx == 0:
        total += sum(list(map(int, value.split(sep='+'))))
    else:
        total -= sum(list(map(int, value.split(sep='+'))))
print(total)
